dsgrid/config/dataset_schema_handler_standard.py

```diff
@@ -0,0 +1,182 @@
+
+        load_data_dimensions = (
+            DimensionType.TIME,
+            config.model.data_schema.load_data_column_dimension,
+        )
+        expected_dimensions = [d for d in DimensionType if d not in load_data_dimensions]
+        if len(dimension_types) != len(expected_dimensions):
+            raise DSGInvalidDataset(
+                f"load_data_lookup does not have the correct number of dimensions specified between trivial and non-trivial dimensions."
+            )
+
+        missing_dimensions = set(expected_dimensions).difference(dimension_types)
+        if len(missing_dimensions) != 0:
+            raise DSGInvalidDataset(
+                f"load_data_lookup is missing dimensions: {missing_dimensions}. If these are trivial dimensions, make sure to specify them in the Dataset Config."
+            )
+
+        for dimension_type in dimension_types:
+            name = dimension_type.value
+            dimension = config.get_dimension(dimension_type)
+            dim_records = dimension.get_unique_ids()
+            lookup_records = get_unique_values(load_data_lookup, name)
+            if dim_records != lookup_records:
+                logger.error(
+                    "Mismatch in load_data_lookup records. dimension=%s mismatched=%s",
+                    name,
+                    lookup_records.symmetric_difference(dim_records),
+                )
+                raise DSGInvalidDataset(
+                    f"load_data_lookup records do not match dimension records for {name}"
+                )
+
+    def _check_dataset_time_consistency(self, config: DatasetConfig, load_data_df):
+        time_dim = config.get_dimension(DimensionType.TIME)
+        time_dim.check_dataset_time_consistency(load_data_df)
+        self._check_dataset_time_consistency_by_id(time_dim, load_data_df)
+
+    def _check_dataset_time_consistency_by_id(self, time_dim, load_data):
+
+        time_ranges = time_dim.get_time_ranges()
+        assert len(time_ranges) == 1, len(time_ranges)
+        time_range = time_ranges[0]
+        # TODO: need to support validation of multiple time ranges: DSGRID-173
+
+        expected_timestamps = time_range.list_time_range()
+        expected_count = len(expected_timestamps)
+
+        timestamps_by_id = (
+            load_data.select("timestamp", "id")
+            .groupby("id")
+            .agg(F.countDistinct("timestamp").alias("distinct_timestamps"))
+        )
+        distinct_counts = timestamps_by_id.select("distinct_timestamps").distinct()
+        if distinct_counts.count() == 1:
+            val = distinct_counts.collect()[0].distinct_timestamps
+            if val != expected_count:
+                raise DSGInvalidDataset(
+                    f"load_data arrays do not have {len(expected_timestamps)} timestamps: actual={row.distinct_timestamps}"
+                )
+            return
+
+        for row in timestamps_by_id.collect():
```

Comment:
I changed the order of this code block and the previous one in a PR based on your comment. Looks like you didn't pick it up.